apollo-ios-codegen/Sources/ApolloCodegenLib/CodegenConfiguration/ApolloCodegenConfiguration.swift

```diff
@@ -268,7 +268,7 @@ public struct ApolloCodegenConfiguration: Codable, Equatable {
       moduleType: ModuleType
     ) {
       self.path = path
-      self.moduleType = moduleType
+      self.moduleType = moduleType == .swiftPackageManager ? .swiftPackage(dependencyType: .default) : moduleType
```

Comment:
Did this to ensure we always convert deprecated `.swiftPackageManager` values to the new `.swiftPackage()` values so the rest of the codebase can just use that (also decoding the deprecated version to the new one from JSON), also reduces the amount of warnings we are generating from using a deprecated case in the code.

Reply:
Oooh, yeah, we can't keep a deprecation warning in the code like this. It will show up in all user's projects. Swift needs better support for suppressing deprecation warnings. It's so frustrating that this still works this way.

There are also issues with deprecating enum cases at all, since switch statements on enums need to be exhaustive, so you always have to switch on the deprecated cases and get warnings.

I almost think it would be better to just add a comment stating the case is deprecated and not using the `@available` to deprecate here for that reason.

Reply:
Yea I could just switch it to doc comment, under the hood we are going to be converting everything to the new value anyways and we can just remove it with the next major release.